easybuild/easyblocks/f/faststructure.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing fastStructure, implemented as an easyblock
+
+@author: Bob Dröge (University of Groningen)
+"""
+import os
+
+from easybuild.easyblocks.generic.cmdcp import CmdCp
+from easybuild.framework.easyconfig import CUSTOM
+
+
+class EB_fastStructure(CmdCp):
+    """Support for building and installing fastStructure."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Change default values of options"""
+        extra = CmdCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for fastStructure."""
+        super(EB_fastStructure, self).__init__(*args, **kwargs)
+
+        self.pyfiles = ['distruct.py', 'chooseK.py', 'structure.py']
+
+    def build_step(self):
+        """Build fastStructure using setup.py."""
+        self.cfg['cmds_map'] = [
+            ('.*', 'cd vars && python setup.py build_ext --inplace && cd .. && python setup.py build_ext --inplace')
+        ]
+        super(EB_fastStructure, self).build_step()
```

Comment:
We can do this better/cleaner:

```python
def build_step(self):
    """..."""
    cwd = change_dir('vars')
    run_cmd("python setup.py build_ext --inplace")
    change_dir(cwd)
    run_cmd("python setup.py build_ext --inplace")
```

That way, if one of both commands fails, you'll know where it failed...